common/edqs/src/main/java/org/thingsboard/server/edqs/data/dp/AbstractDataPoint.java

```diff
@@ -54,4 +54,21 @@ public String toString() {
         return valueToString();
     }
 
+    @Override
+    public int compareTo(DataPoint dataPoint) {
+        String str1 = this.valueToString();
+        String str2 = dataPoint.valueToString();
+
+        if (str1 == null && str2 == null) {
```

Comment:
Use `org.apache.commons.lang3.StringUtils.compareIgnoreCase` instead of your own implementation